### ISSUE TITLE
Zero recall for angular datasets

### ISSUE THREAD

**Issue by ankane** [erikbern/ann-benchmarks#367]
Hi, it looks like recall is always zero for angular datasets like `random-xs-20-angular` and `glove-25-angular` after 399c708544acf169904d67085977022bc31d3d0d.

```sh
git checkout 399c708544acf169904d67085977022bc31d3d0d
export DATASET=random-xs-20-angular
export LIBRARY=annoy
python3 run.py --docker-tag ann-benchmarks-${LIBRARY} --max-n-algorithms 3 --runs 2 --dataset $DATASET --run-disabled --timeout 300
python3 plot.py --dataset $DATASET --output plot.png
```

**Comment by erikbern** [comment]
Explanation of issue is here: https://github.com/erikbern/ann-benchmarks/issues/363#issuecomment-1503160543

Fix: #372
Extra tests: #373

**Comment by ankane** [comment]
Thanks, working great now!
